lib/Sema/TypeCheckStorage.cpp

```diff
@@ -2347,6 +2347,8 @@ PropertyWrapperBackingPropertyInfoRequest::evaluate(Evaluator &evaluator,
         !propertyType->isEqual(expectedPropertyType)) {
       var->diagnose(diag::property_wrapper_incompatible_property,
                     propertyType, wrapperType);
+      var->setInvalid();
+      var->setInterfaceType(ErrorType::get(propertyType));
```

Comment:
I believe `setInvalid` should set interface type to error when appropriate is it right, @CodaFi? What I'm not sure about is should this do `setType(ErrorType::get(propertyType))`?

Reply:
You can't call `setType` on a `VarDecl` (at least not anymore it seems). I suppose I can skip the `setInterfaceType` part because it doesn't make any difference in this case, but I put it there because it's what I've seen being done elsewhere in the codebase. I can remove it though.

Reply:
Sorry, my question is less targeted to your changes, I'm just trying to get a broader sense when `setType(ErrorType)` should or shouldn't be used...  I think you should remove `setInterfaceType` though since that should be done by `setInvalid` I believe (but it's worth a double-check).

Reply:
Oops, you're totally right, it is indeed set by `setInvalid`. I haven't updated my copy of master since a few days and it seems like the [implementation](https://github.com/apple/swift/pull/27922) was changed a few days ago to automatically set the interface type as well.

One difference is that it's not using the var type as the backing type of the `ErrorType`, but none of the current diagnostics have a problem with that, so I guess it's fine.

Reply:
Sounds good!

Reply:
VarDecl’s Type is now a function of its interface type, and invalid is a function of the interface type.  All you have to do nowadays is `setInvalid()`.